vault/proxy.go

```diff
@@ -0,0 +1,296 @@
+package vault
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	socks5 "github.com/armon/go-socks5"
+	isatty "github.com/mattn/go-isatty"
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/crypto/ssh/knownhosts"
+	"golang.org/x/net/http/httpproxy"
+)
+
+type ProxyRouter struct {
+	proxyConf httpproxy.Config
+}
+
+func (n ProxyRouter) Proxy(req *http.Request) (*url.URL, error) {
+	return n.proxyConf.ProxyFunc()(req.URL)
+}
+
+func NewProxyRouter() (*ProxyRouter, error) {
+	httpProxy := getEnvironmentVariable("HTTP_PROXY", "http_proxy")
+	httpsProxy := getEnvironmentVariable("HTTPS_PROXY", "https_proxy")
+
+	allProxy := getEnvironmentVariable("SAFE_ALL_PROXY", "safe_all_proxy")
+	if allProxy != "" {
+		httpProxy = allProxy
+		httpsProxy = allProxy
+	}
+
+	noProxy := getEnvironmentVariable("NO_PROXY", "no_proxy")
+
+	knownHostsFile := getEnvironmentVariable("KNOWN_HOSTS_FILE", "known_hosts_file")
+	skipHostKeyString := getEnvironmentVariable("SKIP_HOST_KEY_VALIDATION", "skip_host_key_validation")
```

Comment:
Are these conventional env vars, or are they specific to safe?  If the latter, they should probably be prefixed with `SAFE_` / `safe_`.

Reply:
👍